pkg/apis/core/validation/validation.go

```diff
@@ -3018,6 +3018,10 @@ func ValidatePodSpec(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, ValidateRuntimeClassName(*spec.RuntimeClassName, fldPath.Child("runtimeClassName"))...)
 	}
 
+	if spec.Overhead != nil {
```

Comment:
I agree this should be done at the runtimeClass admission controller level

Reply:
Definitely makes sense.  Added some basic tests in its place (behind the feature gate).